src/convert.jl

```diff
@@ -225,14 +225,29 @@ end
 function sexp2julia(rl::RList)
     if isdataframe(rl)
         # FIXME add force_missing option to control whether always convert to Union{T, Missing}
-        DataFrame(Any[isa(col, RAltRep) ? sexp2julia(col) : jlvec(col, false) for col in rl.data],
-                  identifier.(names(rl)), makeunique=true)
+        cols = Any[isa(col, RAltRep) ? sexp2julia(col) : jlvec(col, false) for col in rl.data]
+        nms = identifier.(names(rl))
+        obj = DataFrame(cols, nms, makeunique=true)
+        meta = DataAPI.metadata(obj)
+        sizehint!(meta, length(rl.attr))
+        for (key, val) in pairs(rl.attr)
+            key in ("names", "class", "row.names") && continue
```

Comment:
```suggestion
            key in ("names", "class") && continue # skip already processed system attributes
```
I think using *row.names* is a bad practice nowadays, but importing them could be quite convenient, when the user has no control over the contents of RData files.

Reply:
OK. In the original PR all metadata keys where propagated when transforming the data frame, which made importing row names dangerous. But now we can mark them as `:default` style so that they are dropped as soon as the data frame is modified, so it's safe. But see https://github.com/JuliaData/RData.jl/pull/93#discussion_r985217939.